src/rpc/rawtransaction.cpp

```diff
@@ -1760,11 +1776,17 @@ UniValue utxoupdatepsbt(const JSONRPCRequest& request)
 
         const Coin& coin = view.AccessCoin(psbtx.tx->vin[i].prevout);
 
-        std::vector<std::vector<unsigned char>> solutions_data;
-        txnouttype which_type = Solver(coin.out.scriptPubKey, solutions_data);
-        if (which_type == TX_WITNESS_V0_SCRIPTHASH || which_type == TX_WITNESS_V0_KEYHASH || which_type == TX_WITNESS_UNKNOWN) {
+        if (IsSegWitOutput(provider, coin.out.scriptPubKey)) {
             input.witness_utxo = coin.out;
         }
+
+        // Update the inputs using descriptor data.
+        SignPSBTInput(HidingSigningProvider(&provider, /* nosign */ true, /* nobip32derivs */ false), psbtx, i, /* sighash_type */ 1);
```

Comment:
I found using `SignPSBTInput()` a bit jarring here, since nothing is actually being signed. Can you add a comment to say that `SignPSBTInput()` won't actually sign when provided with a `HidingSigningProvider` where `nosign` is true?

Reply:
Added some comments.